tests/unit/data/test_extract_metadata.py

```diff
@@ -54,42 +54,136 @@ def cube_2():
 
     @staticmethod
-    def test_cubelist_title(cubelist_metadata):
+    def test_title_cubelist(cube_1, cube_2):
+        cubelist_metadata = data.extract_metadata.extract_metadata(
+            CubeList([cube_1, cube_2]), 1, [], ['cube'], ['netCDF'], 'title', 'desc')
         assert (cubelist_metadata.title == "title")
 
     @staticmethod
-    def test_total_time_extent(cube_metadata):
+    def test_total_time_extent(cube_1):
+        cube_metadata = data.extract_metadata.extract_metadata(
+            cube_1, 1, [], ['cube'], ['netCDF'])
         assert (np.allclose(cube_metadata.extent.temporal.values, np.linspace(1, 24, 24)))
 
     @staticmethod
-    def test_total_vertical_extent(cube_metadata):
+    def test_total_vertical_extent(cube_1):
+        cube_metadata = data.extract_metadata.extract_metadata(
+            cube_1, 1, [], ['cube'], ['netCDF'])
         assert (cube_metadata.extent.vertical.values == pytest.approx(3.5))
 
     @staticmethod
-    def test_cube_parameters_length(cube_metadata):
+    def test_parameters_length_cube(cube_1):
+        cube_metadata = data.extract_metadata.extract_metadata(
+            cube_1, 1, [], ['cube'], ['netCDF'])
         assert (len(cube_metadata.parameters) == 1)
 
     @staticmethod
-    def test_cubelist_parameters_length(cubelist_metadata):
+    def test_parameters_length_cubelist(cube_1, cube_2):
+        cubelist_metadata = data.extract_metadata.extract_metadata(
+            CubeList([cube_1, cube_2]), 1, [], ['cube'], ['netCDF'], 'title', 'desc')
         assert (len(cubelist_metadata.parameters) == 2)
+
+    @staticmethod
+    def test_containing_polygon_cube(cube_1):
+        cube_metadata = data.extract_metadata.extract_metadata(
+            cube_1, 1, [], ['cube'], ['netCDF'])
+        assert (cube_metadata.extent.spatial.bbox.bounds == (45, -90, 360, 90))
+
+    @staticmethod
+    def test_containing_polygon_equal(cube_1):
+        # two equal polygons
+        cubelist_metadata = data.extract_metadata.extract_metadata(
+            CubeList([cube_1, cube_1]), 1, [], ['cube'], ['netCDF'], 'title', 'desc')
+        assert (cubelist_metadata.extent.spatial.bbox.bounds == (45, -90, 360, 90))
+
+    @staticmethod
+    def test_containing_polygon_overlapping(cube_1, cube_2):
+        # two partially overlapping polygons
+        cubelist_metadata = data.extract_metadata.extract_metadata(
+            CubeList([cube_1, cube_2]), 1, [], ['cube'], ['netCDF'], 'title', 'desc')
+        assert (cubelist_metadata.extent.spatial.bbox.bounds == (1, -90, 360, 100))
+
+    @staticmethod
+    def test_containing_polygon_within(cube_1, cube_3):
+        # one polygon completely within another polygon
+        cubelist_metadata = data.extract_metadata.extract_metadata(
+            CubeList([cube_1, cube_3]), 1, [], ['cube'], ['netCDF'], 'title', 'desc')
+        assert (cubelist_metadata.extent.spatial.bbox.bounds == (-10, -150, 400, 150))
```

Comment:
The brackets are optional here, typically the 'pythonic' way would be to omit them